services/object/src/backends/aws.js

```diff
@@ -0,0 +1,86 @@
+const { Backend } = require('./base');
+const assert = require('assert');
+const aws = require('aws-sdk');
+
+class AwsBackend extends Backend {
+  constructor(options) {
+    super(options);
+
+    for (let prop of ['accessKeyId', 'secretAccessKey', 'bucket', 'signGetUrls']) {
+      assert(prop in options.config, `backend ${options.backendId} is missing ${prop}`);
+    }
+
+    this.config = options.config;
+  }
+
+  async setup() {
+    const credentials = {
+      accessKeyId: this.config.accessKeyId,
+      secretAccessKey: this.config.secretAccessKey,
+    };
+    this.region = await getBucketRegion({ bucket: this.config.bucket, credentials });
+    this.s3 = new aws.S3({ region: this.region, ...credentials });
+  }
+
+  async temporaryUpload(object, data) {
+    await this.s3.putObject({
+      Bucket: this.config.bucket,
+      Key: object.name,
+      Body: data,
+    }).promise();
+  }
+
+  async availableDownloadMethods(object) {
+    return ['simple'];
+  }
+
+  async downloadObject(object, method, params) {
+    switch (method){
+      case 'simple': {
+        let url;
+        if (this.config.signGetUrls) {
+          url = await this.s3.getSignedUrlPromise('getObject', {
```

Comment:
Signed URLs stretch my concept of ``simple``, but it is retrievable with a ``GET``. So stretch but not break.